plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -337,12 +353,28 @@ private void setFieldValue(String path, Object value, boolean append) {
         }
     }
 
-    public String getMetadata(MetaData metaData) {
-        return this.metaData.get(metaData.getFieldName());
+    public String getEsMetadata(MetaData esMetadata) {
+        return this.esMetadata.get(esMetadata.getFieldName());
     }
 
-    public void setMetaData(MetaData metaData, String value) {
-        this.metaData.put(metaData.getFieldName(), value);
+    public Map<String, String> getEsMetadata() {
```

Comment:
I made the change suggested by Martijn. I think ingest metadata does need to be in a different map to prevent field name conflicts, unless we change approach regarding how we store _source and metadata fields in a single map, but that looks good at the moment I think.